server/src/main/java/org/opensearch/gateway/remote/ClusterMetadataManifest.java

```diff
@@ -1121,7 +1121,7 @@ private static String componentPrefix(Object[] fields) {
             PARSER.declareString(ConstructingObjectParser.constructorArg(), INDEX_NAME_FIELD);
             PARSER.declareString(ConstructingObjectParser.constructorArg(), INDEX_UUID_FIELD);
             PARSER.declareString(ConstructingObjectParser.constructorArg(), UPLOADED_FILENAME_FIELD);
-            PARSER.declareString(ConstructingObjectParser.constructorArg(), COMPONENT_PREFIX_FIELD);
+            PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), COMPONENT_PREFIX_FIELD);
```

Comment:
there is no versioning for UploadedMetadata?

Reply:
If you mean codec version then this version is part of the manifest file which contains the UploadedMetadata.

Reply:
versioning for UploadedMetadata itself if we introduce new fields or remove existing ones

Reply:
We have added codec version for objects which have their own file like manifest and index metadata. In those cases we can first read the codec version from the file and then deserialize accordingly.
In this case, if we add codec version it will be serialized along with the object within the manifest. We will not know the version until we deserialize it.

Reply:
Updated the logic to use manifest codec version to decide ser/de.
Manifest has codec v0, v1 and v2.
Here we need to add parser for v0 and v2. V2 parser will be used for codec v2 and above. For codec below v2, v0 parser will be used.